File: code/backend/src/main/java/com/ioc/internship/service/AIService.java
package com.ioc.internship.service;

import com.ioc.internship.dto.request.AiChatRequest;
import com.ioc.internship.dto.response.AiChatResponse;
import com.ioc.internship.dto.response.AiConversationResponse;
import com.ioc.internship.dto.response.AiMessageResponse;
import com.ioc.internship.entity.UserEntity;

import java.util.List;

public interface AIService {
    AiChatResponse chat(AiChatRequest request, UserEntity currentUser);
    List<AiConversationResponse> getConversations(UserEntity currentUser);
    AiConversationResponse getConversation(Long id, UserEntity currentUser);
    List<AiMessageResponse> getMessages(Long conversationId, UserEntity currentUser);
    void archiveConversation(Long id, UserEntity currentUser);
    void restoreConversation(Long id, UserEntity currentUser);
    void deleteConversation(Long id, UserEntity currentUser);
}

File: code/frontend/src/services/ai-service.ts
import api from "@/lib/axios";

export interface AiChatRequest {
  content: string;
  conversationId?: number;
}

export interface AiChatResponse {
  response: string;
  conversationId: number;
  conversationTitle: string;
}

export interface AiConversationResponse {
  id: number;
  title: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface AiMessageResponse {
  id: number;
  role: string;
  content: string;
  createdAt: string;
}

export const aiService = {
  chat: async (request: AiChatRequest): Promise<AiChatResponse> => {
    const res = await api.post("/ai/chat", request);
    return res.data;
  },

  getConversations: async (): Promise<AiConversationResponse[]> => {
    const res = await api.get("/ai/conversations");
    return res.data;
  },

  getConversation: async (id: number): Promise<AiConversationResponse> => {
    const res = await api.get(`/ai/conversations/${id}`);
    return res.data;
  },

  getMessages: async (id: number): Promise<AiMessageResponse[]> => {
    const res = await api.get(`/ai/conversations/${id}/messages`);
    return res.data;
  },

  archiveConversation: async (id: number): Promise<void> => {
    const res = await api.post(`/ai/conversations/${id}/archive`);
    return res.data;
  },

  restoreConversation: async (id: number): Promise<void> => {
    const res = await api.post(`/ai/conversations/${id}/restore`);
    return res.data;
  },

  deleteConversation: async (id: number): Promise<void> => {
    const res = await api.delete(`/ai/conversations/${id}`);
    return res.data;
  }
};
